types/xovis/PC2S/uplink.js

```diff
@@ -1,94 +1,108 @@
-
-  if (event.payload != undefined) {
-    for (var i = 0; i < event.payload.length; i++) {
-      var data = {};
-
-      if (event.payload[i].type === "LineCount") {
-        if (event.payload[i].direction === 'forward') {
-          data = { "fw": 1, "bw": 0 };
+  // Xovis Event
+  if (event.values !== undefined) {
+    for (let i = 0; i < event.values.length; i++) {
+      let data = {};
+
+      if (event.values[i].type === "LineCount") {
+        if (event.values[i].direction === "forward") {
+          data = { fw: 1, bw: 0 };
         } else {
-          data = { "fw": 0, "bw": 1 };
+          data = { fw: 0, bw: 1 };
         }
 
         // Mask
-        var faceMask = event.payload[i].object.faceMask;
+        const { faceMask } = event.values[i].object;
         if (faceMask !== undefined) {
-          if (faceMask == "NO_MASK") {
+          if (faceMask === "NO_MASK") {
             data.faceMask = 0;
           } else {
             data.faceMask = 1;
           }
         }
 
         // Gender
-        var gender = event.payload[i].object.gender;
+        const { gender } = event.values[i].object;
         if (gender !== undefined) {
-          if (gender == "FEMALE") {
+          if (gender === "FEMALE") {
             data.gender = "f";
           } else {
             data.gender = "m";
           }
         }
 
-        emit('sample', { data: data, topic: "line_count" });
-
-      } else if (event.payload[i].type === "ZoneEntry" || event.payload[i].type === "ZoneExit") {
-        //data.id = event.payload[i].object.id; // Can check on Person ID to see how long individual persons take time 
-        //data.timestamp =  event.payload[i].timestamp; // dwelltime is the same as timein-timeout
-        emit('sample', { data: data, topic: event.payload[i].type, save: false }); // Not saved gets used in Rule
-
-      } else if (event.payload[i].type === "ZoneDwellTime") {
-        emit('sample', { data: { "dwelltime": Math.round(event.payload[i].dwellTime / 1000) }, topic: "dwelltime" });
+        emit("sample", { data, topic: "line_count" });
+      } else if (
+        event.values[i].type === "ZoneEntry" ||
+        event.values[i].type === "ZoneExit"
+      ) {
+        // data.id = event.payload[i].object.id; // Can check on Person ID to see how long individual persons take time
+        // data.timestamp =  event.payload[i].timestamp; // dwelltime is the same as timein-timeout
+        // emit("sample", { data, topic: event.values[i].type, save: false }); // Not saved gets used in Rule
+      } else if (event.values[i].type === "ZoneDwellTime") {
+        emit("sample", {
+          data: { dwelltime: Math.round(event.values[i].dwellTime / 1000) },
+          topic: "dwelltime",
+        });
       }
     }
 
-    // Xovis Line Count -- Without Domain
-  } else if (event.data !== undefined) {
-
-    var data = event.data.content;
-    var rawSample = {};
+    // Xovis Line Count
+  } else if (event.content !== undefined) {
```

Comment:
event.content will never exist I believe in the new version

Reply:
i passed the messages on akenza.io without a devicetype and used these as tests. So it should work.

Reply:
still it will not exist therefore this code makes no sense

Reply:
?
`{
  "sensor-time": {
    "timezone": "Europe/Zurich",
    "time": "2021-09-23T13:50:04+02:00"
  },
  "status": {
    "code": "OK"
  },
  "content": {
    "element": [
      {
        "element-id": 0,
        "element-name": "Line 0",
        "sensor-type": "SINGLE_SENSOR",
        "data-type": "LINE",
        "from": "2021-09-23T13:49:00+02:00",
        "to": "2021-09-23T13:50:00+02:00",
        "resolution": "ONE_MINUTE",
        "measurement": [
          {
            "from": "2021-09-23T13:49:00+02:00",
            "to": "2021-09-23T13:50:00+02:00",
            "value": [
              {
                "value": 0,
                "label": "fw"
              },
              {
                "value": 0,
                "label": "bw"
              }
            ]
          }
        ]
      }
    ]
  },
  "sensor-info": {
    "serial-number": "80:1F:12:D5:30:DC",
    "ip-address": "192.168.0.94",
    "name": "Office Entrance",
    "group": "Akenza Office",
    "device-type": "PC2S - UL"
  }
}`

Reply:
where did you get this example?